WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift

```diff
@@ -0,0 +1,89 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireAPISupport
+import WireDataModel
+import WireDataModelSupport
+import XCTest
+
+@testable import WireDomain
+
+final class UserPushRemoveEventProcessorTests: XCTestCase {
+
+    var sut: UserPushRemoveEventProcessor!
+
+    var coreDataStack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+
+    var context: NSManagedObjectContext {
+        coreDataStack.syncContext
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        coreDataStack = try await coreDataStackHelper.createStack()
+        sut = UserPushRemoveEventProcessor(
+            repository: UserRepository(
+                context: context,
+                usersAPI: MockUsersAPI()
+            )
```

Comment:
Shouldn't we test with a mock user repository?

Reply:
in theory yes but in practice processors are mostly just passthrough objects to the repository (they don't do much except forwarding data to the repo). 
So if we pass mock repository to processors, what we can test is that repo methods are correctly invocated.

It makes me wonder if we should even test processors since repositories are already tested at the WireDomain level and they perform most of the work.

Reply:
resolving this conversation after aligning on how we test processors that is: injecting the mock repository and ensuring that repo methods are correctly invocated.